actors/common/src/main/java/org/sunbird/common/ShadowUserProcessor.java

```diff
@@ -0,0 +1,361 @@
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+
+    private List<String> getOrganisationIds(Map<String, Object> dbUser) {
+        List<String> organisationsIds = new ArrayList<>();
+        ((List<Map<String, Object>>) dbUser.get(JsonKey.ORGANISATIONS)).stream().forEach(organisation -> {
+            organisationsIds.add((String) organisation.get(JsonKey.ORGANISATION_ID));
+        });
+        return organisationsIds;
+    }
+
+
+
+
+    private void syncUserToES(String userId) {
+        Map<String, Object> fullUserDetails = Util.getUserDetails(userId, null);
+        try {
+            Future<Boolean> future = elasticSearchService.update(JsonKey.USER, userId, fullUserDetails);
+            if ((boolean) ElasticSearchHelper.getResponseFromFuture(future)) {
+                ProjectLogger.log("ShadowUserMigrationScheduler:updateUserStatus: data successfully updated to elastic search with userId:".concat(userId + ""), LoggerEnum.INFO.name());
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            ProjectLogger.log("ShadowUserMigrationScheduler:syncUserToES: data failed to updates in elastic search with userId:".concat(userId + ""), LoggerEnum.ERROR.name());
+        }
+    }
+
+
+    private void deleteUserOrganisations(Map<String, Object> esUserMap) {
+        ((List<Map<String, Object>>) esUserMap.get(JsonKey.ORGANISATIONS)).stream().forEach(organisation -> {
+            String id = (String) organisation.get(JsonKey.ID);
+            deleteOrgFromUserOrg(id);
+        });
+    }
+
+    private void deleteOrgFromUserOrg(String id) {
+        Response response = cassandraOperation.deleteRecord(JsonKey.SUNBIRD, JsonKey.USER_ORG, id);
+        ProjectLogger.log("ShadowUserProcessor:deleteOrgFromUserOrg:user org is deleted ".concat(response.getResult() + ""));
+    }
+
+
+    private void registerUserToOrg(String userId, String organisationId) {
+        Map<String, Object> reqMap = new WeakHashMap<>();
+        List<String> roles = new ArrayList<>();
+        roles.add(ProjectUtil.UserRole.PUBLIC.getValue());
+        reqMap.put(JsonKey.ROLES, roles);
+        String hashTagId = Util.getHashTagIdFromOrgId(organisationId);
+        reqMap.put(JsonKey.HASHTAGID, hashTagId);
+        reqMap.put(JsonKey.ID, ProjectUtil.getUniqueIdFromTimestamp(1));
+        reqMap.put(JsonKey.USER_ID, userId);
+        reqMap.put(JsonKey.ORGANISATION_ID, organisationId);
+        reqMap.put(JsonKey.ORG_JOIN_DATE, ProjectUtil.getFormattedDate());
+        reqMap.put(JsonKey.IS_DELETED, false);
+        Util.DbInfo usrOrgDb = Util.dbInfoMap.get(JsonKey.USR_ORG_DB);
+        try {
+            cassandraOperation.insertRecord(usrOrgDb.getKeySpace(), usrOrgDb.getTableName(), reqMap);
+        } catch (Exception e) {
+            ProjectLogger.log(e.getMessage(), e);
+        }
+    }
+
+    private void createUserExternalId(String userId, ShadowUser shadowUser) {
```

Comment:
createUserExternalId will provide you data and then write one more method as saveUserExternalId to save that data.

Reply:
done